src/components/IndexTable/README.md

```diff
@@ -0,0 +1,1244 @@
+  - list of collections
+  - product listings list
+  - channel lists
+  - resource list attributes
+  - list attributes
+  - exceptions list
+  - list secondary actions
+  - secondary actions in a list
+  - list of resources
+  - filter
+  - sort
+---
+
+# Index table
+
+An index table displays a collection of objects of the same type, like orders or products. The main job of an index table is to help merchants get an at-a-glance of the objects to perform actions or navigate to a full-page representation of it.
+
+Index tables can also:
+
+- Support [customized index rows and columns](https://polaris.shopify.com/components/lists-and-tables/resource-item)
+- Include bulk actions so merchants can act on multiple objects at once
+- Support sorting and [filtering](https://polaris.shopify.com/components/lists-and-tables/filters) of long lists
+- Be paired with pagination to make long lists digestible
+
+---
+
+## Examples
+
+### Simple index table
+
+A index table with simple items and no bulk actions, sorting, or filtering.
+
+TODO
+
+```jsx
+function SimpleIndexTableExample() {
+  const customers = [
+    {
+      id: '3411',
+      url: 'customers/341',
+      name: 'Mae Jemison',
+      location: 'Decatur, USA',
+      orders: 20,
+      amountSpent: '$2,400',
+    },
+    {
+      id: '2561',
+      url: 'customers/256',
+      name: 'Ellen Ochoa',
+      location: 'Los Angeles, USA',
+      orders: 30,
+      amountSpent: '$140',
+    },
+  ];
+  const resourceName = {
+    singular: 'customer',
+    plural: 'customers',
+  };
+
+  const {
+    selectedResources,
+    allResourcesSelected,
+    handleSelectionChange,
+  } = useIndexResourceState(customers);
+
+  const rowMarkup = customers.map(
+    ({id, name, location, orders, amountSpent}, index) => (
+      <IndexTable.Row
+        id={id}
+        key={id}
+        selected={selectedResources.includes(id)}
+        position={index}
+      >
+        <IndexTable.Cell first>
+          <TextStyle variation="strong">{name}</TextStyle>
+        </IndexTable.Cell>
+        <IndexTable.Cell>{location}</IndexTable.Cell>
+        <IndexTable.Cell>{orders}</IndexTable.Cell>
+        <IndexTable.Cell>{amountSpent}</IndexTable.Cell>
+      </IndexTable.Row>
+    ),
+  );
+
+  return (
+    <Card>
+      <IndexProvider
+        resourceName={resourceName}
+        itemCount={customers.length}
+        selectedItemsCount={
+          allResourcesSelected ? 'All' : selectedResources.length
+        }
+        onSelectionChange={handleSelectionChange}
+      >
+        <IndexTable
+          headings={[
+            {title: 'Name'},
+            {title: 'Location'},
+            {title: 'Order count'},
+            {title: 'Amount spent'},
+          ]}
+        >
+          {rowMarkup}
+        </IndexTable>
+      </IndexProvider>
+    </Card>
+  );
+}
+```
+
+### Simple small screen index table
+
+A small screen index table with simple items and no bulk actions, sorting, or filtering.
```

Comment:
I wonder if rather than having a separate example like this that looks strange when viewed on larger screens, we instead have one example that actually does the responsive switch between the views. I think that would be more helpful to consumers who are trying to wire this up.

Reply:
@umar-k Is adding responsive examples [here](https://github.com/Shopify/polaris-react/pull/3951) 🎉